src/QsFmt/SPECIFICATION.md

```diff
@@ -0,0 +1,58 @@
+﻿# Q# Formatter Design Specification Document
+
+QsFmt is a source code formatter for Q#.
+
+## Where to get the formatter
+
+The `Microsoft.Quantum.QsFormatter` NuGet package will be distributed with the rest of Q# on NuGet.org.
+As a dotnet tool, it is installed with the command `dotnet tool install Microsoft.Quantum.QsFormatter`.
+
+## How to use the command line formatter
+
+After installing the tool the command `qsfmt` is used to run the formatter.
+There are two commands supported for the tool: 'format' and 'update'.
+The 'format' command runs the tool as a formatter for Q# code, running only those transformation rules that update whitespace and affect indentation. The underlying code will not be changed.
+The 'update' command allows the tool to be used as a means of updating old Q# code, replacing deprecated syntax with newer supported syntax. These transformation rules change the actual code given to them, and may occasionally fail.
+These two commands work separately and can't be run together. If the user wishes to both format and update their code, they would need to run the tool twice, first with 'update' and then with 'format'.
+
+The tool will initially only support the 'update' command, as the formatting functionality is still being worked on. The 'format' command will be unavailable until it is supported.
```

Comment:
Is there any default if the command ('update', 'format') is not specified?

Reply:
I think, maybe in the future, the `format` command will be the default, but because we are not initially shipping with the `format` command, I think for now we go with no default. The tool will say it is missing a command if none is given. I don't think we want the `update` to be the default.